07/AcyclicGraph.scala

```diff
@@ -0,0 +1,214 @@
+import scala.io.Source
+
+object AcyclicGraph {
+case class Worker(step: Step, startTick: Int)
+
+case class Dependency(name: String, satisfied: Boolean)
+
+  case class Step(name: String, dependencies: List[Dependency]) {
+    private val duration: Int = name match {
+      case "A" => 61
+      case "B" => 62
+      case "C" => 63
+      case "D" => 64
+      case "E" => 65
+      case "F" => 66
+      case "G" => 67
+      case "H" => 68
+      case "I" => 69
+      case "J" => 70
+      case "K" => 71
+      case "L" => 72
+      case "M" => 73
+      case "N" => 74
+      case "O" => 75
+      case "P" => 76
+      case "Q" => 77
+      case "R" => 78
+      case "S" => 79
+      case "T" => 80
+      case "U" => 81
+      case "V" => 82
+      case "W" => 83
+      case "X" => 84
+      case "Y" => 85
+      case "Z" => 86
+      case _ => 0
+      }
+
+    def isReady(): Boolean = {
+      def go(dependencies: List[Dependency]): Boolean = {
+        if (dependencies.isEmpty) true
+        else if (dependencies.head.satisfied) go(dependencies.tail)
+        else false
+      }
+      go(this.dependencies)
```

Comment:
There are `forall` and `exists` combinators on collections. Take a look and you might find them useful.